File: herencia-dto/AMarotoChivite-HerenciaDTO/AMaroto-Chivite-HerenciasDto-Java/src/main/java/Ficheros/HerenciasDto/models/dto/PersonDTO.java
package Ficheros.HerenciasDto.models.dto;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

public abstract sealed class PersonDTO {

    public abstract String getName();

    public abstract int getEdad();

    public abstract String getType();

    @Root(name = "alumno")
    public static final class Alumno extends PersonDTO {

        @Attribute(name = "type")
        private final String type = "Alumno";
        @Element(name = "nombreAlumno")
        private String name;
        @Element(name = "edad")
        private String edad;

        public Alumno() {
            // Constructor sin argumentos requerido para SimpleXML
        }

        public Alumno(
                @Element(name = "nombreAlumno") String name,
                @Element(name = "edad") String edad
        ) {
            this.name = name;
            this.edad = edad;
        }

        @Override
        public String getName() {
            return name;
        }

        public String setName(String name) {
            this.name = name;
            return name;
        }

        @Override
        public int getEdad() {
            return Integer.parseInt(edad);
        }

        public String setEdad(String edad) {
            this.edad = edad;
            return edad;
        }

        @Override
        public String getType() {
            return type;
        }

        public String setType(String type) {
            return type;
        }

        @Override
        public String toString() {
            return "Alumno(name='" + name + "', edad=" + edad + ")";
        }
    }

    @Root(name = "profesor")
    public static final class Profesor extends PersonDTO {

        @Attribute(name = "type")
        private final String type = "Profesor";
        @Element(name = "nombreProfesor")
        private String name;
        @Element(name = "edad")
        private String edad;
        @Element(name = "modulo")
        private String modulo;

        public Profesor() {
            // Constructor sin argumentos requerido por SimpleXML
        }

        public Profesor(
                @Element(name = "nombreProfesor") String name,
                @Element(name = "edad") String edad,
                @Element(name = "modulo") String modulo
        ) {
            this.name = name;
            this.edad = edad;
            this.modulo = modulo;
        }

        @Override
        public String getName() {
            return name;
        }

        public String setName(String name) {
            this.name = name;
            return name;
        }

        @Override
        public int getEdad() {
            return Integer.parseInt(edad);
        }

        public String setEdad(String edad) {
            this.edad = edad;
            return edad;
        }

        @Override
        public String getType() {
            return type;
        }

        public String setType(String type) {
            return type;
        }

        public String getModulo() {
            return modulo;
        }

        @Override
        public String toString() {
            return "Profesor(name='" + name + "', edad=" + edad + ", modulo='" + modulo + "')";
        }
    }
}

File: herencia-dto/AMarotoChivite-HerenciaDTO/AMarotoChivite-HerenciasDto-Kotlin/src/main/kotlin/Ficheros/HerenciasDto/models/dto/PersonDTO.kt
package Ficheros.HerenciasDto.models.dto

import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

sealed class PersonDTO {
    abstract val name: String
    abstract val edad: String
    abstract val type: String

    @Root(name = "alumno") // Para XML
    data class Alumno(
        @field:Element(name = "nombreAlumno") // Para XML escritura
        @param:Element(name = "nombreAlumno")   // Para XML lectura
        override val name: String,
        @field:Element(name = "edad")
        @param:Element(name = "edad")
        override val edad: String,
        @field:Attribute(name = "type")
        @param:Attribute(name = "type")
        override val type: String = "Alumno"
    ) : PersonDTO() {

        override fun toString(): String {
            return "Alumno(name='$name', edad=$edad)"
        }
    }
    
    @Root(name = "profesor") // Para XML
    data class Profesor(
        @field:Element(name = "nombreProfesor") // Para XML escritura
        @param:Element(name = "nombreProfesor") // Para XML lectura
        override val name: String,
        @field:Element(name = "edad")
        @param:Element(name = "edad")
        override val edad: String,
        @field:Element(name = "modulo")
        @param:Element(name = "modulo")
        val modulo: String,
        @field:Attribute(name = "type")
        @param:Attribute(name = "type")
        override val type: String = "Profesor"
    ) : PersonDTO() {

        override fun toString(): String {
            return "Profesor(name='$name',edad=$edad, modulo='$modulo')"
        }
    }
}


